New Features

brunovarelaAptoide opened the pull request:
- Created getSkuDetailsAsync
- Created ConsumeAsync
- Created LaunchBillingFlow
- Enum SkuType
- Project keys hidden
- Toolbox main activity prepared for testing

JoaoRaimundo committed:
Merge pull request #2 from Aptoide/dev

Merge changes from Dev

paulotrezentos committed:
Update README.md

paulotrezentos committed:
Update README.md

brunovarelaAptoide committed:
Initial Commit

- Created package appcoins_billing and android_appcoins_billing

brunovarelaAptoide committed:
Communication Working

- Created a factory
- Communication is working

brunovarelaAptoide committed:
IsbillingSupported

brunovarelaAptoide committed:
isBillingSupported

The function is billing supported is working for V3 Version

brunovarelaAptoide committed:
Subscrition V5 e V3 Validation

Added a v5 e v3 subscrition verifications.

brunovarelaAptoide committed:
Added methods

Added methods to Interface AppcoinsCatupultBilling.

brunovarelaAptoide committed:
Created SkuDetails

Created the class SKUDetails
Changed compile tag to API

mikieribeiro committed:
Purchase class created

JsonObject altered in this class in order to make it work

brunovarelaAptoide committed:
Inventory Class

Commiting Inventory Class

brunovarelaAptoide committed:
getSkuDetailsAsync

Created the functionallity of get SKUDetailsAsync and isBillingSupported.
The Client used for the test is in ToolBox MainActivity class.
Still need some tweaks with error handling.

mikieribeiro committed:
QueryPurchase method created

QueryPurchase method created but currently being tested with hamered code

brunovarelaAptoide committed:
Creation of the buyIntent and launchFlow methods

- Created the buy intent
- Created the launchflow

mikieribeiro committed:
QueryPurchase APPC-857

QueryPurchase method now validating all purchases

brunovarelaAptoide committed:
Few changes to work testing

- Changes on the toolbox.
- Changes on the configurations

brunovarelaAptoide committed:
Merge pull request #8 from Aptoide/APPC-856-Create-the-getBuyIntent-method

Appc 856 create the get buy intent method

mikieribeiro committed:
Purchase Result

Purchases being printed on the user side (token and sku)

mikieribeiro committed:
Merge branch 'Create-Catapult-billing-SDK' into Create-ConsumePurchase-APPC-858

mikieribeiro committed:
Merge pull request #7 from Aptoide/Create-ConsumePurchase-APPC-858

Create QueryPurchase appc 858

brunovarelaAptoide committed:
Correction of merge between Branchs

brunovarelaAptoide committed:
First Commit

-Organized classes in packages.

Trinkes committed:
[purchases/APPC-857] - implement queryPurchases flow in the SDK

brunovarelaAptoide committed:
Listenner Added

Added a Listener to startService

brunovarelaAptoide committed:
Changed AppCoinsBillingStateListenner

onBillingServiceDisconnected added to the AppCoinsBillingStateListenner

brunovarelaAptoide committed:
refactoring

AppCoinsBillingStateListenner refactored to AppCoinsBillingStateListener

mikieribeiro committed:
Merge remote-tracking branch 'origin/dev' into feature/APPC-857-query-puchases

Trinkes committed:
correct dependency and sdk versions

brunovarelaAptoide committed:
New Features

- Created getSkuDetailsAsync
- Created ConsumeAsync
- Created LaunchBillingFlow
- Enum SkuType
- Project keys hidden
- Toolbox main activity prepared for testing

mikieribeiro committed:
APPC-847

-Changes in the PayloadHelper made
- OrderReference, DeveloperPayload and Origin added to LaunchBillingFlow

brunovarelaAptoide committed:
Merge branch 'feature/APPC-857-query-puchases' into APPC-855-856-858

brunovarelaAptoide committed:
Changed String Values

String Values

brunovarelaAptoide committed:
Merge branch 'APPC-855-856-858' of https://github.com/Aptoide/bds-sdk into APPC-855-856-858

Trinkes commented on android_appcoins_billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/CatapultAppcoinsBilling.java:
```diff
@@ -18,27 +29,32 @@ public PurchasesResult queryPurchases(String skuType) {
+      activity.startIntentSenderForResult(pendingIntent.getIntentSender(),
+          billingFlowParams.getRequestCode(), new Intent(), 0, 0, 0);
+    } catch (IntentSender.SendIntentException e) {
+      e.printStackTrace();
```
you have to deliver this errors to the developer

Trinkes commented on android_appcoins_billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/helpers/AndroidBillingMapper.java:
```diff
@@ -68,4 +72,58 @@ public PurchasesResult map(Bundle bundle, String skuType) {
+        description);
+  }
+
+  public HashMap<String, Object> mapBundleToHashMapGetIntent(Bundle bundle) {
```
don't use Maps with objects to return values, create a data class instead

Trinkes commented on gradle.properties:
```diff
@@ -21,9 +21,9 @@ APPCOINS_IAB_PERMISSION=com.appcoins.BILLING
-APPCOINS_DEV_IAB_BIND_PACKAGE=com.appcoins.wallet.dev
-APPCOINS_DEV_IAB_PERMISSION=com.appcoins.dev.BILLING
-APPCOINS_DEV_IAB_KEY=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAuTiKetzlcfeMOEFKFjKmCJbV6tpmMXFTActnlwXkMCcS3SGLt1zCy9MX2zyIS8yFK4EhVzRZpSwM4/5dQ3I+iAa6DQu1N/vf1UFuQ0dtdH8+KkunjnElJWq/KmgMKlkSnT4OKaOQJ0T7tR/Qs8lMLVb7XspK2NQfIPYpyFH8bYnh2Gdl9AMBTnPSxetviG9gu0GLzI5svm3CoYlIPAfQWkGooIxNd8XH5iehyhcnlNfgVdHbf7nQ5Sifr+9dFQqbBD21VNR6gshGdFUXo5Zx4K49p792H2eE8bpFMugPlimbIlA3fWsSN2A3PUoBi8cua6/udQPFf9yiWS ...
+APPCOINS_DEV_IAB_BIND_PACKAGE=replace
```
this is needed

Trinkes commented on gradle.properties:
```diff
@@ -21,9 +21,9 @@ APPCOINS_IAB_PERMISSION=com.appcoins.BILLING
+APPCOINS_DEV_IAB_BIND_PACKAGE=replace
+APPCOINS_DEV_IAB_PERMISSION=replace
+APPCOINS_DEV_IAB_KEY=replace
 APPCOINS_DEV_IAB_WALLET_ADDR=0x75e8BdCf84B9bd585753dB64A8637A52Bd46f851
```
this should be a placeholder

Trinkes commented on android_appcoins_billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/helpers/IabHelper.java:
```diff
@@ -8,28 +8,22 @@
-
-public class IabHelper implements ServiceConnection {
 
+public class IabHelper{
```
delete this class

Trinkes commented on android_appcoins_billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/CatapultAppcoinsBilling.java:
```diff
@@ -18,27 +29,32 @@ public PurchasesResult queryPurchases(String skuType) {
 
-  public void launchPurchaseFlow(Object act, String sku, String itemType, List<String> oldSkus,
-      int requestCode, ResponseListener listener, String extraData) {
+  public void lauchBillingFlow(Activity activity, BillingFlowParams billingFlowParams) {
```
typo

brunovarelaAptoide committed:
Commit changes from Code review

- Changed Values to gradle .properties
- Create Data class to substitute the HashMap Returns

mikieribeiro committed:
APPC-847

-Public key added to AppCoinsBilling constructor

brunovarelaAptoide committed:
Changes to PullRequest

-Removed some unusued classes
-Removed propreties to be replace in gradle

mikieribeiro committed:
APPC-847

-verifyPurchase method created
-Security class moved to java
-java.util.Base64 imported instead of android.util.Base64
-code adapted to new import

mikieribeiro committed:
Merge pull request #13 from Aptoide/APPC-847-VerifyPurchase

Appc 847 verify purchase

brunovarelaAptoide committed:
changed build.gradle

change the min api version

Trinkes commented on android_appcoins_billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/CatapultAppcoinsBilling.java:
```diff
@@ -18,27 +30,36 @@ public PurchasesResult queryPurchases(String skuType) {
 
-  public void launchPurchaseFlow(Object act, String sku, String itemType, List<String> oldSkus,
-      int requestCode, ResponseListener listener, String extraData) {
+  public void launchBillingFlow(Activity activity, BillingFlowParams billingFlowParams)
```
does not respect google's interface

Trinkes commented on android_appcoins_billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/CatapultAppcoinsBilling.java:
```diff
@@ -18,27 +30,36 @@ public PurchasesResult queryPurchases(String skuType) {
   }
-  */
+
   public void startService(final AppCoinsBillingStateListener listener) {
```
change startService and stopService methods name to startConnection and endConnection(my bad), following google's interface

Trinkes commented on appcoins_billing/src/main/java/com/appcoins/sdk/billing/AppCoinsBilling.java:
```diff
@@ -1,32 +1,73 @@
+      return pr;
     } catch (ServiceConnectionException e) {
       e.printStackTrace();
       return new PurchasesResult(Collections.emptyList(), -1);
```
return the correct error code. Check it here: https://developer.android.com/reference/com/android/billingclient/api/BillingClient.BillingResponse

Trinkes commented on android_appcoins_billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/helpers/AppCoinsAndroidBillingRepository.java:
```diff
@@ -39,9 +44,59 @@ public AppCoinsAndroidBillingRepository(int apiVersion, String packageName,
     try {
       Bundle purchases = service.getPurchases(apiVersion, packageName, skuType, null);
-      return mapper.map(purchases, skuType);
+      PurchasesResult pr = mapper.map(purchases, skuType);
```
use meaningful names for variables

Trinkes commented on android_appcoins_billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/service/WalletBillingService.java:
```diff
@@ -1,60 +1,52 @@
+    return null;
+  }
+
+  public Bundle getBuyIntentToReplaceSkus(int apiVersion, String packageName, List<String> oldSkus,
```
not used

Trinkes commented on appcoins_billing/src/main/java/com/appcoins/sdk/billing/AppCoinsBilling.java:
```diff
@@ -1,32 +1,73 @@
+        String purchaseData = purchase.getOriginalJson();
+        String signature = purchase.getSignature();
+
+        try {
```
don't use nested try catches

Trinkes commented on appcoins_billing/src/main/java/com/appcoins/sdk/billing/AppCoinsBilling.java:
```diff
@@ -1,32 +1,73 @@
+  }
 
+  @Override public void consumeAsync(String purchaseToken, ConsumeResponseListener listener) {
+    ConsumeAsync consumeAsync = new ConsumeAsync(purchaseToken, listener, repository);
```
This is not async, it will execute at the same thread

Trinkes commented on appcoins_billing/src/main/java/com/appcoins/sdk/billing/AppCoinsBilling.java:
```diff
@@ -1,32 +1,73 @@
   @Override
-  public void launchPurchaseFlow(Object act, String sku, String itemType, List<String> oldSkus,
-      int requestCode, ResponseListener listener, String extraData) {
+  public HashMap<String, Object> launchBillingFlow(BillingFlowParams params, String payload)
```
dont use maps as return types

Trinkes commented:
review all error codes according with the google's

Trinkes commented on android_appcoins_billing/src/main/java/com/appcoins/sdk/android_appcoins_billing/helpers/AppCoinsAndroidBillingRepository.java:
```diff
@@ -39,9 +44,59 @@ public AppCoinsAndroidBillingRepository(int apiVersion, String packageName,
+    }
+  }
+
+  @Override public int consumeAsync(String purchaseToken) throws ServiceConnectionException {
```
Is this method async?

Trinkes commented on appcoins_billing/src/main/java/com/appcoins/sdk/billing/Security.java:
```diff
@@ -0,0 +1,113 @@
+      throws IllegalArgumentException, RuntimeException {
+    try {
+
+      byte[] decodedKey = Base64.getMimeDecoder()
```
our min sdk is 21, you need an alternative

Trinkes commented on toolbox/build.gradle:
```diff
@@ -1,21 +1,20 @@
     applicationId "com.appcoins.trivialdrivesample"
-    targetSdkVersion COMPILE_SDK_VERSION
-    minSdkVersion MIN_SDK_VERSION
+    minSdkVersion 25
```
our min sdk is 21

Trinkes replied:
use project's default values

mikieribeiro committed:
New Features PR

-Made sugested changes

Trinkes commented on toolbox/build.gradle:
```diff
@@ -1,42 +1,114 @@
+    debug {
       minifyEnabled false
       proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
+//      applicationIdSuffix ".debug"
```
remove commented code

Trinkes commented on toolbox/build.gradle:
```diff
@@ -1,42 +1,114 @@
-    minSdkVersion MIN_SDK_VERSION
-    targetSdkVersion TARGET_SDK_VERSION
+  defaultConfig {
+    applicationId "com.appcoins.trivialdrivesample"
```
revert application id change

Trinkes commented on toolbox/src/main/java/com/asf/appcoins/toolbox/Application.java:
```diff
@@ -0,0 +1,30 @@
+
+    public final static String developerAddress = "0x75e8BdCf84B9bd585753dB64A8637A52Bd46f851";
+
+    // public final static String developerAddress = "0x5fe8f9d5e7bab4b94fa441b871683d124da270f1";
```
remove commented code

Trinkes commented on appcoins_billing/src/main/java/com/appcoins/sdk/billing/ErrorCode.java:
```diff
@@ -0,0 +1,31 @@
+  ErrorCode(int value) {
+    this.value = value;
+  }
+
```
review why do we need a map here

brunovarelaAptoide committed:
Chages

- Rename package name
- Remove comments

mikieribeiro committed:
APPC-855-856-858

-Added exception errors accordingly to google play's billing response

Trinkes commented on appcoins-billing/src/main/java/com/appcoins/sdk/billing/ConsumeAsync.java:
```diff
@@ -15,15 +15,15 @@ public ConsumeAsync(String token, ConsumeResponseListener listener, Repository r
       listener.onConsumeResponse(response, token);
     } catch (ServiceConnectionException e) {
-      e.printStackTrace();
+     ResponseCode.SERVICE_UNAVAILABLE.getValue();
```
what is this code supposed to do?

Trinkes commented on toolbox/build.gradle:
```diff
@@ -1,42 +1,113 @@
 
 android {
-  compileSdkVersion COMPILE_SDK_VERSION
+  compileSdkVersion 28
```
why are you hardcoding this value while we have constants for that already?

Trinkes commented on android-appcoins-billing/build.gradle:
```diff
@@ -0,0 +1,87 @@
+    flavorDimensions "iab"
+
+    defaultConfig {
+        minSdkVersion 21
```
use constants we have already

brunovarelaAptoide committed:
Code Review Changes

Changed the .gradle files to used pre established variables.
Add a response that was not used.

brunovarelaAptoide committed:
changed build files

-Removed unusued properties from build grad

brunovarelaAptoide committed:
Cleanning properties

- Removed unused properties from build.gradle from toolbox.

Trinkes committed:
add isReady and endConnection methods
extracted an interface to externally

brunovarelaAptoide committed:
Code review changes

- Changed the response processing.

Trinkes committed:
Merge remote-tracking branch 'origin/dev' into APPC-855-856-858
